File: 프로그래머스/2/42584. 주식가격/주식가격.py
def solution(prices):
    answer = []
    n=len(prices)
    for i in range(n):
        c=0
        for j in range(i+1,n):
            if prices[j]<prices[i]:
                c+=1
                break
            c+=1
        answer.append(c)
    return answer

File: 프로그래머스/2/42584. 주식가격/주식가격.java
import java.util.Stack;

class Solution {
    public int[] solution(int[] prices) {
        int n = prices.length;
        int[] answer = new int[n];
        
        for(int i = 0; i < n; i++) {
            answer[i] = n - i - 1;
        }
        
        Stack<Integer> stack = new Stack<>();
        stack.push(0);
        for(int i = 1; i < n; i++) {
            while (!stack.empty() && prices[stack.peek()] > prices[i]) {
                int j = stack.pop();
                answer[j] = i - j;
            }
            stack.push(i);
        }
        
        return answer;
    }
}